### PR TITLE
Update Release Managers (promotions, new members, retirement, and inactivity updates)

### PR DESCRIPTION
**Pull request by justaugustus**
#### What type of PR is this:

/kind cleanup documentation

#### What this PR does / why we need it:

We've got a set of team changes in flight, which are captured here.
This PR will only contain changes to the Release Managers page; I'll follow-up with OWNERS updates in future PRs.

- Rotate leadership team listed on Release Managers page (https://github.com/kubernetes/community/pull/5286)
  - @tpepper steps down as SIG Chair (https://github.com/kubernetes/sig-release/issues/1321)
  - @saschagrunert steps up as SIG Chair (https://github.com/kubernetes/sig-release/issues/1322)
  - @hasheddan steps up as SIG Technical Lead (https://github.com/kubernetes/sig-release/issues/1322)

- Promote @puerco to full-fledged Release Manager (ref: https://groups.google.com/a/kubernetes.io/g/release-managers-private/c/TeZm5Zxxwz4)

  Adolfo has been doing excellent work with the Release Engineering tooling, consistently exercises good judgement, and ability to execute.

- Add @ameukam as a Release Manager Associate (discussed w/ @kubernetes/sig-release-leads)
  
  Arnaud continues to be involved in WG K8s Infra, has assisted with standing up our instance of Triage Party, and is working on migrating our CI build jobs to community infrastructure.
  As we prepare to take on more responsibilities around GCS/GCR artifact administration and K8s Infra, his participation in the Release Managers groups will be strategic.

- @tpepper, @hoegaarden, and @dougm stepping down as Release Managers (results of the Release Managers check-in survey)

---

The final commit for this PR will be the removal of inactive Release Managers and Associates, which is gated on responses to the Release Managers check-in survey (which closes this Friday, November 13, EOD US Eastern).
/hold

The following notices were sent about the survey:
- 11/4: https://groups.google.com/a/kubernetes.io/g/release-managers/c/-0THUc578s8/m/WjFcf7h3BwAJ
- 11/4: https://kubernetes.slack.com/archives/C2C40FMNF/p1604504049103400
- 11/10: https://kubernetes.slack.com/archives/C2C40FMNF/p1605018614180000

~At present, 5 of our 18 current Release Managers and Associates have not completed the survey.
Individual pings have been sent to each of these group members.~

All Release Managers and Associates have checked in.

/assign @saschagrunert @hasheddan 
cc: @kubernetes/release-engineering

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
/lgtm

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by justaugustus** [comment]
We've hit the lazy consensus deadline on this leadership rotation and the timeout for the Release Managers check-in!
/hold

**Comment by justaugustus** [comment]
LOL:
/hold cancel

**Comment by justaugustus** [comment]
Sorry y'all, tagged you on the wrong PR:
/unassign @mrbobbytables @cblecker

**Pull request merged**
